os/arch/arm/src/common/up_restoretask.c

```diff
@@ -71,26 +74,63 @@ void up_restoretask(struct tcb_s *tcb)
 #endif
 
 		/* Restore the MPU registers in case we are switching to an application task */
-#ifdef CONFIG_ARM_MPU
 #ifdef CONFIG_APP_BINARY_SEPARATION
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+		if (g_umm_app_id) {
+			*g_umm_app_id = tcb->app_id;
+		}
+#endif
+
+#ifdef CONFIG_ARCH_USE_MMU
+#if 0
+		uint32_t ctxidr = mmu_get_asid();
+		uint8_t asid = ctxidr & 0xFF;
+		if (asid != tcb->app_id) {
+			mmu_switch_ttbr_asid((uint32_t)tcb->pgtbl, tcb->app_id);			
+		}
+#endif
+
+  irqstate_t flags;
+      flags = enter_critical_section();
+		lldbg("Write TTBR = 0x%08x\n", tcb->pgtbl);
+		cp15_wrttb((uint32_t)tcb->pgtbl);
+		cp15_invalidate_tlbs();
```

Comment:
```

static inline void cp15_invalidate_tlbs(void)
{
  __asm__ __volatile__
    (
      "\tdsb\n"
#ifdef CONFIG_ARM_HAVE_MPCORE
      "\tmcr p15, 0, r0, c8, c3, 0\n" /* TLBIALLIS */
      "\tmcr p15, 0, r0, c7, c1, 6\n" /* BPIALLIS */
#else
```
We may also need to invalidate the instruction cache.

```
mcr p15, 0, r0, c7, c1, 0
```

Reply:
I thought so too. We can try it.